src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -79,7 +80,19 @@ public Task<InitializeResult> Initialize(JToken input, CancellationToken cancell
         }
 
         [JsonRpcMethod(Methods.InitializedName)]
-        public Task Initialized() => Task.CompletedTask;
+        public async Task Initialized()
+        {
+            // Publish diagnostics for all open documents immediately following initialization.
+            var openDocuments = _workspace.GetOpenDocumentIds();
+            foreach (var documentId in openDocuments)
+            {
+                var document = _workspace.CurrentSolution.GetDocument(documentId);
+                if (document != null)
+                {
+                    await PublishDiagnosticsAsync(_workspace.CurrentSolution, document).ConfigureAwait(false);
```

Comment:
continually referencing _workspace.CurrentSolution is very scary.  it could change over time.  Instead, pull this out above hte foreach look and always reference the same instance.  Passing that instance into `PublishDiagnosticsAsync` shouldn't be necessary either.  it's redundant given that `document` is passed in.